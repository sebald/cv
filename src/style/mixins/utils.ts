import { CSSProperties } from 'react';
import { StyledMixin } from '../theme';

export const propOrEmpty = <P extends keyof CSSProperties>(
  name: P
): StyledMixin<Pick<CSSProperties, P>> => (props: Pick<CSSProperties, P>) => {
  const p = props[name];
  return p ? { [name]: p } : {};
};
