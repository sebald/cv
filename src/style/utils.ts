import { theme } from './theme';

const { htmlFontSize } = theme;

export const px = (value: number) => `${value}px`;

export const pxToRem = (pixel: number, base = htmlFontSize) =>
  `${parseFloat((pixel / base).toFixed(5))}rem`;

export const remToPx = (rem: string, base = htmlFontSize) =>
  parseFloat(rem) * base;

export const ensurePx = (value: number | string, base = htmlFontSize) =>
  typeof value === 'string' ? remToPx(value, base) : value;
