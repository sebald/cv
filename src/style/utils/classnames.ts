export const mergeClassNames = (...names: (string | undefined)[]) => ({
  className: names
    .filter(x => typeof x === 'string')
    .join(' ')
    .trim(),
});
