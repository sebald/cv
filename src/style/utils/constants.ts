export const htmlFontSize = 16;

export const ratio = {
  perfectFourth: 1.333,
  perfectFifth: 1.5,
  goldenRatio: 1.618,
  majorSixth: 1.667,
};
