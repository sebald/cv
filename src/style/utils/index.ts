export * from './classnames';
export * from './constants';
export * from './createMixinComponent';
export * from './interpolation';
export * from './unit';
export * from './typography';
