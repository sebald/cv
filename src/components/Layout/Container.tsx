import { media, styled } from '../../style';

export const Container = styled.div`
  display: flex;
  flex-direction: column;

  height: 100vh;
  width: 100%;

  ${media('small')`
    flex-wrap: wrap;
  `};

  @media print {
    flex-direction: row;
    flex-wrap: wrap;
  }
`;
