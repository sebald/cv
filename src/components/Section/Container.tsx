import * as React from 'react';

import { px, styled } from '../../style';
import { SectionHeading } from './SectionHeading';
import { SectionContainerProps } from './types';

const ContainerWrapper = styled.section`
  max-width: ${(props) => px(props.theme.layout.content)};
`;

export const Container: React.FC<SectionContainerProps> = ({
  className,
  title,
  children,
}) => (
  <ContainerWrapper className={className}>
    <SectionHeading>{title}</SectionHeading>
    {children}
  </ContainerWrapper>
);

Container.displayName = 'SectionContainer';
