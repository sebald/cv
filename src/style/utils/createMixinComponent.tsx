import React from 'react';

import { styled } from '../styled-components';
import { StyledMixin } from '../theme';

import { mergeClassNames } from './classnames';
import { WithClassName } from './types';

/**
 * Idea stolen from:
 * https://github.com/jxnblk/grid-styled/tree/master/styled-space
 */
const StyledChildren: React.FC<WithClassName> = ({ className, children }) => {
  const styledChildren = React.Children.toArray(children).map((child) =>
    React.isValidElement<any>(child)
      ? React.cloneElement(
          child,
          mergeClassNames(child.props.className, className)
        )
      : child
  );

  return <>{styledChildren}</>;
};

export const createMixinComponent = <P extends {}>(
  ...mixins: StyledMixin<P>[]
) => styled(StyledChildren)([] as any, ...mixins);
