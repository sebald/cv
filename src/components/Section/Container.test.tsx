import * as React from 'react';

import { shallowWithTheme } from '../../test.utils';
import { Container } from './Container';

test('render correct', () => {
  const wrapper = shallowWithTheme(<Container title="Title" />);
  expect(wrapper).toMatchSnapshot();
});

test('render children', () => {
  const wrapper = shallowWithTheme(<Container title="Title">child</Container>);
  expect(wrapper).toMatchSnapshot();
});
