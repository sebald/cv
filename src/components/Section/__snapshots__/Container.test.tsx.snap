// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render children 1`] = `
<styled.section>
  <Component>
    Title
  </Component>
  child
</styled.section>
`;

exports[`render correct 1`] = `
<styled.section>
  <Component>
    Title
  </Component>
</styled.section>
`;
